pkg/storage/bucket/client.go

```diff
@@ -101,6 +108,14 @@ func (cfg *Config) Validate() error {
 		}
 	}
 
+	if cfg.StoragePrefix != "" {
+		// Takes from S3's guide on key names https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html#object-key-guidelines
+		acceptablePrefixCharacters := regexp.MustCompile(validPrefixCharactersRegex)
```

Comment:
Notice that `..` is allowed, but it may not work correctly when we're joining path (eg. for filesystem bucket, it would get out of bucket root!)

Reply:
that's a good point, I will add it as validation

Reply:
but so would a single segment `.` (`path.Clean(path.Join(".", "/", "."))` results in `.` https://go.dev/play/p/xnPVYnmeFKY). I am not sure I can explain what exactly is allowed, so I will link to the docs of `path.Clean` and explain that path.Clean should be a noop on the prefix

Reply:
but even that doesn't cut it... they can still provide `..` as the path... I think this is becoming too complex. Pushed some changes in 7d74f12435905f37d871bb2d21a1f6ac45d3f10f and lets' discuss this in the [other comment thread](https://github.com/grafana/mimir/pull/1686#discussion_r861563909)